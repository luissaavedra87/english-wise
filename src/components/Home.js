import React from 'react';
import { Link } from 'react-router-dom';
import teacherSalon from '../resources/teacher-salon.jpg';

const Home = () => {
  const text = (
    <h3>
      Welcome back to English Wise
    </h3>
  );

  return (
    <>
      <div className="home">
        <div>{text}</div>
        <img src={teacherSalon} alt="teacher-salon" />
        <div>
          If you are looking for a personal learning,&nbsp;
          search between our teachers and schedule a 1 hour class.
        </div>
      </div>
      <Link to="/schedule">User Schedule</Link>
    </>
  );
};

export default Home;
