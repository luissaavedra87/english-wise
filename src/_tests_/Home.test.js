import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../components/Home';

afterEach(cleanup);

it('renders ok', () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});

it('check for logout text', () => {
  const { getByTestId } = render(<Home />);
  expect(getByTestId('welcome-text')).toHaveTextContent('Welcome back to English Wise');
});

it('check for p text', () => {
  const { getByTestId } = render(<Home />);
  expect(getByTestId('info-text')).toHaveTextContent('If you are looking for a personal learning, search between our teachers and schedule a one hour class.');
});
