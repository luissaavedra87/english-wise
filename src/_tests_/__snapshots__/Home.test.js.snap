// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<DocumentFragment>
  <div
    class="home mx-auto"
  >
    <div
      data-testid="welcome-text"
    >
      <h4
        class="d-flex justify-content-center my-3"
      >
        Welcome back to English Wise
      </h4>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <img
        alt="teacher-salon"
        src="teacher-salon.jpg"
      />
    </div>
    <div
      class="home-text mx-auto"
      data-testid="info-text"
    >
      If you are looking for a personal learning,Â  search between our teachers and schedule a one hour class.
    </div>
  </div>
</DocumentFragment>
`;
